### PR TITLE
Update test_panel.py

### LINKED ISSUE #12101: DEPR/TEST: inspect deprecation warning
hmm, this looks pretty new. should fix this code.

```
...../home/travis/build/pydata/pandas/pandas/tests/test_panel.py:201: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() instead
```

build is [here](https://travis-ci.org/pydata/pandas/jobs/103526993). This is a numpy-dev build, need to check out the other builds.

### PR DESCRIPTION
**Pull request by ranarag**
closes #12101

changed deprecated getargspec to signature for python 3 while keeping the getargspec for python 2

### CODE REVIEW

File: pandas/tests/test_panel.py [hunk 1]
```diff
@@ -2,11 +2,14 @@
 # pylint: disable=W0612,E1101
 
 from datetime import datetime
-from inspect import getargspec
+try:
+  from inspect import getargspec
```
Comment (jreback) [inline pandas/tests/test_panel.py]: let's add this to ``pandas/compat/__init__`` and then import from there

File: pandas/tests/test_panel.py [hunk 2]
```diff
@@ -2,11 +2,14 @@
 # pylint: disable=W0612,E1101
 
 from datetime import datetime
-from inspect import getargspec
+try:
+  from inspect import signature
```
Comment (jreback) [inline pandas/tests/test_panel.py]: this needs to go in ``pandas/compat/__init__.py``

Comment (ranarag) [inline pandas/tests/test_panel.py]: I have tried to do that in  Update __init__.py #12117 .
Do I need to make any other changes?

**Comment by jreback** [comment]
the compat changes need to be incorporated in this PR

File: pandas/tests/test_panel.py [hunk 3]
```diff
@@ -2,11 +2,14 @@
...
 
 from datetime import datetime
-from inspect import getargspec
 import operator
 import nose
 from functools import wraps
-
+try:
+  from inspect import signature
+except:
```
Comment (jreback) [inline pandas/tests/test_panel.py]: should be in ``compat/__init__.py``
``from inspect import getargspec as signature``for PY2
``from inspect import signature`` for PY3

then just ``from compat import signature`` and use it

Comment (ranarag) [inline pandas/tests/test_panel.py]: Thank you sir for helping me out. :smiley:

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/0b8fa9dcdb6f47e40441467dd7ebb529914bb719

thanks!

**Pull request closed without merging**